### PR TITLE
Device type tracking

### PR DESCRIPTION
**Pull request by quarterto**
/cc @commuterjoy. Need to specify more media queries. As well as screen size it would probably be useful to track orientation and screen density.

### CODE REVIEW

File: src/analytics/device-type.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+var DeviceType = function (o) {
+    this.breakpoints = o.breakpoints || {
+        desktop: '(min-width: 1200px)'
+    };
+};
+
+// Fire a beacon with media matches
+DeviceType.prototype.track = function (root) { 
+    this.mediaMatches(root, function(matches) {
+        dispatchCustomEvent('beacon:devicetype', matches);
```
Comment (commuterjoy) [inline src/analytics/device-type.js]: I don't think this should be another tracking call. 

Device class should be a property of the user model ... https://github.com/Financial-Times/next-beacon-component/blob/master/src/models/user.js#L10

**Comment by quarterto** [comment]
Probably slightly unintended use case: we can use this to detect when the user prints an article, using the `print` media query!

**Comment by quarterto** [comment]
The callback gets fired multiple times when multiple media queries trigger, we should probably not do that.

**Comment by commuterjoy** [comment]
PR looks good to me - add some more break points & make sure it's only added as a property on user.js

**Comment by quarterto** [comment]
> make sure it's only added as a property on user.js

What do you mean by that?

**Comment by commuterjoy** [comment]
```js
    return {
        version: '1',
        erights: new User(cookie).id() || 'unknown',
        isStaff: new User(cookie).isStaff(),
        flags: new User(cookie).flags(),
        ua: navigator.userAgent,
        deviceClass: 'mobile'   <------- you need to populate a single value here
    };
```

**Comment by quarterto** [comment]
Closing in favour of #23.

**Pull request closed without merging**